ui/css/treeherder.css

```diff
@@ -218,9 +218,29 @@ th-watched-repo {
     transition: width 0.2s;
 }
 
+#platform-job-text-search-field-parent {
+    position: relative;
+}
+
 #platform-job-text-search-field:focus,
 #platform-job-text-search-field:valid {
     width: 300px !important;
+    padding-right: 20px;
+}
+
+#platform-job-text-search-field:valid + #filter-clear-button {
+    display: inherit !important;
```

Comment:
This is some pretty neat css. I guess if the psuedo selector isn't met then the implied display value is none?

From what I can tell in local testing, the `!important` doesn't appear to be required.

In case another means of manipulating the display is of interest, we use ng-show/hide with a scope variable for the [add related bugs field](https://github.com/mozilla/treeherder/blob/d5ae8deb9f041ce538eee1a5d8d05c60f09e56be/ui/partials/main/thPinboardPanel.html#L26). Not sure if that would help facilitate the desired enter behavior.